devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -50,63 +61,152 @@ YarpRobotLoggerDevice::~YarpRobotLoggerDevice() = default;
 {
-    constexpr auto logPrefix = "[YarpRobotLoggerDevice][setupRobotSensorBridge]";
+    constexpr auto logPrefix = "[YarpRobotLoggerDevice::setupExogenousInputs]";
 
-    auto bridgeConfig = config.findGroup("RobotSensorBridge");
-    if (bridgeConfig.isNull())
+    auto ptr = params.lock();
+    if (ptr == nullptr)
     {
-        log()->error("{} Missing required group 'RobotSensorBridge'", logPrefix);
+        log()->info("{} No exogenous input will be logged.", logPrefix);
+        return true;
+    }
+
+    std::vector<std::string> inputs;
+    if (!ptr->getParameter("exogenous_inputs", inputs))
+    {
+        log()->error("{} Unable to get the exogenous inputs.", logPrefix);
         return false;
     }
 
-    auto originalHandler = std::make_shared<YarpImplementation>();
-    originalHandler->set(bridgeConfig);
+    for (const auto& input : inputs)
+    {
+        auto group = ptr->getGroup(input).lock();
+        std::string portName, signalName;
+        if (group == nullptr || !group->getParameter("port_name", portName)
+            || !group->getParameter("signal_name", signalName))
+        {
+            log()->error("{} Unable to get the parameters related to the input: {}.",
+                        logPrefix,
+                        input);
+            return false;
+        }
+
+        if (!m_exogenousPorts[signalName].open(portName))
+        {
+            log()->error("{} Unable to open the port named: {}.", logPrefix, portName);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+bool YarpRobotLoggerDevice::setupTelemetry(
+    std::weak_ptr<const ParametersHandler::IParametersHandler> params, const double& devicePeriod)
+{
+    constexpr auto logPrefix = "[YarpRobotLoggerDevice::setupTelemetry]";
+
+    auto ptr = params.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} The parameters handler is not valid.", logPrefix);
+        return false;
+    }
+
+    yarp::telemetry::experimental::BufferConfig config;
+    config.filename = "robot_logger_device";
+    config.auto_save = true;
+    config.save_periodically = true;
+    config.file_indexing = "%Y_%m_%d_%H_%M_%S";
+    config.mat_file_version = matioCpp::FileVersion::MAT7_3;
+
+    if (!ptr->getParameter("save_period", config.save_period))
+    {
+        log()->error("{} Unable to get the 'save_period' parameter for the telemetry.", logPrefix);
+        return false;
+    }
+
+    // the telemetry will flush the content of its storage every config.save_period
+    // and this device runs every devicePeriod
+    // so the size of the telemetry buffer must be at least config.save_period / devicePeriod
+    // to be sure we are not going to lose data the buffer will be 10% longer
+    constexpr double percentage = 0.1;
+    config.n_samples = static_cast<int>(ceil((1 + percentage) //
+                                             * (config.save_period / devicePeriod)));
```

Comment:
```suggestion
    config.n_samples = static_cast<int>(std::ceil((1 + percentage) //
                                             * (config.save_period / devicePeriod)));
```

available in <cmath>.